pkg/mock_doc.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2021 Security Scorecard Authors
```

Comment:
Use `mockgen` to generate mocks - https://github.com/golang/mock

Reply:
for a simple interFace like the one I have, is it worth using it? I'm genuinely asking because i've never use `mockgen` and I don't know how it would help me here. Note that the implementation is fairly simple in my case and took 15mn to write.

Reply:
The biggest advantage being that it provides far more flexible mock options (like checking number of times a fn is called, `DoAndReturn` on a call etc.). It'll also automate extending the mock if the interface grows in the future.